src/sentry/rules/history/preview.py

```diff
@@ -377,29 +391,64 @@ def apply_frequency_conditions(
             raise PreviewException
         conditions.append(condition_cls(project, data=condition_data))
 
-    filtered_activity = defaultdict(list)
-    for group, activities in group_activity.items():
-        # TODO: only EventFrequencyCondition is supported right now, so we can reuse the same query
-        buckets = get_frequency_buckets(project, start, end, group, dataset_map[group])
-        pass_count = [0] * len(activities)
-        for condition in conditions:
-            for i, activity in enumerate(activities):
-                try:
-                    if condition.passes_activity_frequency(activity, buckets):
-                        pass_count[i] += 1
-                except NotImplementedError:
-                    raise PreviewException
-
-        for i in range(len(activities)):
-            if (
-                pass_count[i]
-                and condition_match == "any"
-                or pass_count[i] == len(conditions)
-                and condition_match == "all"
-            ):
-                filtered_activity[group].append(activities[i])
-
-    return filtered_activity
+    if condition_match == "all":
+        filtered_activity = {}
+        for group, activities in group_activity.items():
+            # TODO: only EventFrequencyCondition is supported right now, so we can reuse the same query
+            buckets = get_frequency_buckets(
+                project, start, end, group, dataset_map.get(group, None)
+            )
+            if not has_issue_state_condition:
+                # If there are no issue state change conditions, then we won't have any initial activities to base our frequency condition queries off of. Instead, we take the first frequency condition and create the initial activities from that
+                for bucket in buckets:
+                    activity = ConditionActivity(
+                        group, ConditionActivityType.FREQUENCY_CONDITION, bucket["roundedTime"]
+                    )
+                    try:
+                        if conditions[0].passes_activity_frequency(activity, buckets):
+                            activities.append(activity)
+                    except NotImplementedError:
+                        raise PreviewException
+
+            passes = [True] * len(activities)
+            for condition in conditions if has_issue_state_condition else conditions[1:]:
+                for i, activity in enumerate(activities):
+                    try:
+                        if passes[i] and not condition.passes_activity_frequency(activity, buckets):
+                            passes[i] = False
+                    except NotImplementedError:
+                        raise PreviewException
+
+            filtered_activity[group] = [activities[i] for i in range(len(activities)) if passes[i]]
+
+        return filtered_activity
+    elif condition_match == "any":
+        # Find buckets that pass at least one condition, and create condition activity from it
```

Comment:
This was wrongly implemented with the `pass_count` array, I forgot that issue state conditions and freq conditions are under the same "any" match